src/arithmetic_impls.rs

```diff
@@ -0,0 +1,294 @@
+use crate::{decimal::CalculationResult, ops, Decimal};
+use core::ops::{Add, Div, Mul, Rem, Sub};
+use num_traits::{CheckedAdd, CheckedDiv, CheckedMul, CheckedRem, CheckedSub};
+
+// Macros and `Decimal` implementations
+
+macro_rules! impl_checked {
+    ($(#[$($doc:meta)*])* $fun:ident, $impl:ident) => {
+        /// Like the checked version but returns lower or upper bounds instead of `None`.
```

Comment:
I think this should be under `impl_saturated` right?

Reply:
Ops, my bad